add lab 4 solutuon

aapakhomov opened the pull request.

aapakhomov committed:
add lab 4 solutuon

dvorotnev commented on cvlib/src/corner_detector.cpp:
```diff
@@ -16,15 +16,64 @@ cv::Ptr<corner_detector_fast> corner_detector_fast::create()
+    for (const auto& offset : offsets)
+    {
+        const auto offset_pixel = img.at<uint8_t>(point + offset);
+        if (offset_pixel < pixel - threshold || offset_pixel > pixel + threshold) ++count;
```
Нужно, чтобы пиксели проходили **одну и ту же проверку**. Т.е. либо N пикселей **подряд** светлее, либо N пикселей **подряд** темнее.

dvorotnev commented on cvlib/src/corner_detector.cpp:
```diff
@@ -16,15 +16,64 @@ cv::Ptr<corner_detector_fast> corner_detector_fast::create()
+    int count = 0;
+    const int threshold = 30;
+    const auto pixel = img.at<uint8_t>(point);
+    for (const auto& offset : offsets)
```
Когда просматриваются все пиксели нужно учесть ещё, что они расположены на окружности, а не на линии. Т.е. проверка должна зацикливаться.

Пример:
Допустим pixel = 0, threshold = 9, а значения на окружности:
10 10 10 10 10 10 2 2 2 2 10 10 10 10 10 10

aapakhomov committed:
исправление замечаний:
1. итератор дважды проходит по вектору смещений (окружности)
2. изменена логика проверки - пиксели проходят одну и ту же проверку

dvorotnev commented:
:+1: можно мёржить, оценка 8 баллов
